CompanionAppService/CompanionAppService.cs

```diff
@@ -680,22 +680,32 @@ public static List<String> ProhibitedCommoditiesFromProfile(dynamic json)
         }
 
         // Obtain the list of commodities from the profile
-        private static List<CommodityMarketQuote> CommodityQuotesFromProfile(dynamic json)
+        private static List<CommodityMarketQuote> CommodityQuotesFromProfile(JObject json)
         {
             List<CommodityMarketQuote> quotes = new List<CommodityMarketQuote>();
 
             if (json["lastStarport"] != null && json["lastStarport"]["commodities"] != null)
             {
-                foreach (dynamic commodity in json["lastStarport"]["commodities"])
+                foreach (JObject commodity in json["lastStarport"]["commodities"])
                 {
                     CommodityDefinition commodityDef = CommodityDefinition.CommodityDefinitionFromEliteID((long)commodity["id"]);
+                    if (commodityDef == null || (string)commodity["name"] != commodityDef.edname)
```

Comment:
If we fail to lookup a commodity by EliteId, we report that data to Rollbar. In the meantime, I believe we can create a basic commodity definition from the info available in the Frontier API rather than discarding that commodity completely. Thoughts?

Reply:
I considered that, but now that we have that spreadsheet from FD we should have all the known commodities, whether common, rare or xeno, So after consideration I have gone back to a more defensive approach.

Open to persuasion on this one though.

Reply:
We have a one-time spreadsheet with no guarantee of similar future spreadsheets. And when we don't have a definition, we want to make that as invisible as possible to the end user. So I think it's worthwhile to create a definition wherever we can.

Reply:
After reflection, I'd like to ship an iteration where we don't report these to Rollbar, if only to see what else Rollbar can tell us if we get through a month without blowing our limit. Armed with that knowledge, we can decide if we want to bring it back.

Reply:
This isn't what is blowing limits on Rollbar. 
Right now the biggest item on Rollbar is exceptions obtaining Frontier API data. https://rollbar.com/EDCD/EDDI/items/3927/?item_page=0&#instances